Null object reference on app start

From `user.getId()` on DashboardActivity

<img width="488" alt="Screenshot 2019-06-06 at 10 52 34 PM" src="https://user-images.githubusercontent.com/2836302/59071390-edb08880-88ad-11e9-885a-5df2c5d219e3.png">

` java.lang.RuntimeException: Unable to start activity ComponentInfo{org.ole.planet.myplanet/org.ole.planet.myplanet.ui.dashboard.DashboardActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.ole.planet.myplanet.model.RealmUserModel.getId()' on a null object reference
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2325)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.ActivityThread.access$800(ActivityThread.java:151)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:102)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.os.Looper.loop(Looper.java:135)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:5254)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at java.lang.reflect.Method.invoke(Method.java:372)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.ole.planet.myplanet.model.RealmUserModel.getId()' on a null object reference
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at org.ole.planet.myplanet.ui.dashboard.DashboardActivity.onCreate(DashboardActivity.java:75)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.Activity.performCreate(Activity.java:5990)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet W/System.err: 	... 10 more
06-06 22:48:13.372 9163-9163/org.ole.planet.myplanet D/OLE: log: Handle exception Unable to start activity ComponentInfo{org.ole.planet.myplanet/org.ole.planet.myplanet.ui.dashboard.DashboardActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.ole.planet.myplanet.model.RealmUserModel.getId()' on a null object reference
06-06 22:48:13.403 9163-9173/? I/art: Debugger is no longer active`

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -7,8 +7,8 @@ android {
         applicationId "org.ole.planet.myplanet"
         minSdkVersion 19
         targetSdkVersion 27
-        versionCode 387
-        versionName "0.3.87"
+        versionCode 388
+        versionName "0.3.88"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true
         multiDexEnabled true

--- a/app/src/main/java/org/ole/planet/myplanet/ui/dashboard/DashboardActivity.java
+++ b/app/src/main/java/org/ole/planet/myplanet/ui/dashboard/DashboardActivity.java
@@ -73,6 +73,11 @@ public class DashboardActivity extends DashboardElementActivity implements OnHom
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         user = new UserProfileDbHandler(this).getUserModel();
+        if(user  == null){
+            Utilities.toast(this, "Session expired.");
+            logout();
+            return;
+        }
         if (user.getId().startsWith("guest")){
             getTheme().applyStyle(R.style.GuestStyle, true);
         }

--- a/app/src/main/java/org/ole/planet/myplanet/ui/dashboard/DashboardActivity.java
+++ b/app/src/main/java/org/ole/planet/myplanet/ui/dashboard/DashboardActivity.java
@@ -72,15 +72,7 @@ public class DashboardActivity extends DashboardElementActivity implements OnHom
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        user = new UserProfileDbHandler(this).getUserModel();
-        if(user  == null){
-            Utilities.toast(this, "Session expired.");
-            logout();
-            return;
-        }
-        if (user.getId().startsWith("guest")){
-            getTheme().applyStyle(R.style.GuestStyle, true);
-        }
+        checkUser();
         setContentView(R.layout.activity_dashboard);
         mTopToolbar = findViewById(R.id.my_toolbar);
         bellToolbar = findViewById(R.id.bell_toolbar);
@@ -112,6 +104,18 @@ public class DashboardActivity extends DashboardElementActivity implements OnHom
 
         openCallFragment((PreferenceManager.getDefaultSharedPreferences(this).getBoolean("bell_theme", false)) ?
                 new BellDashboardFragment() : new DashboardFragment());
+    }
+
+    private void checkUser() {
+        user = new UserProfileDbHandler(this).getUserModel();
+        if(user  == null){
+            Utilities.toast(this, "Session expired.");
+            logout();
+            return;
+        }
+        if (user.getId().startsWith("guest")){
+            getTheme().applyStyle(R.style.GuestStyle, true);
+        }
     }
 
     private void topbarSetting() {

--- a/app/src/main/res/values/versions.xml
+++ b/app/src/main/res/values/versions.xml
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="app_version">0.3.87</string>
+    <string name="app_version">0.3.88</string>
 </resources>

